superset-frontend/src/components/CronPicker/CronPicker.tsx

```diff
@@ -110,22 +110,86 @@ export const CronPicker = styled((props: CronProps) => (
     <ReactCronPicker locale={LOCALE} {...props} />
   </ConfigProvider>
 ))`
-  .react-js-cron-field {
-    margin-bottom: 0px;
-  }
-  .react-js-cron-select:not(.react-js-cron-custom-select) > div:first-of-type,
-  .react-js-cron-custom-select {
-    border-radius: ${({ theme }) => theme.gridUnit}px;
-    background-color: ${({ theme }) =>
-      theme.colors.grayscale.light4} !important;
-  }
-  .react-js-cron-custom-select > div:first-of-type {
-    border-radius: ${({ theme }) => theme.gridUnit}px;
-  }
-  .react-js-cron-custom-select .ant-select-selection-placeholder {
-    flex: auto;
-  }
-  .react-js-cron-custom-select .ant-select-selection-overflow-item {
-    align-self: center;
-  }
+  ${({ theme }) => css`
+    :has(.react-js-cron-months) {
+      display: grid !important;
+      grid-template-columns: repeat(2, 50%);
+      column-gap: ${theme.gridUnit}px;
+      row-gap: ${theme.gridUnit * 2}px;
+      div:has(.react-js-cron-hours) {
+        grid-column: span 2;
+        display: flex;
+        justify-content: space-between;
+        .react-js-cron-field {
+          width: 50%;
+        }
+      }
+    }
+
+    :not(:has(.react-js-cron-months)) {
+      display: grid;
+      grid-template-columns: repeat(2, 50%);
+      column-gap: ${theme.gridUnit}px;
+      row-gap: ${theme.gridUnit * 2}px;
+      .react-js-cron-period {
+        grid-column: span 2;
+      }
+      div:has(.react-js-cron-hours) {
+        grid-column: span 2;
+        display: flex;
+        justify-content: space-between;
+        .react-js-cron-field {
+          width: 50%;
+        }
+      }
+    }
+
+    :not(:has(.react-js-cron-month-days)) {
+      .react-js-cron-week-days {
+        grid-column: span 2;
+      }
+    }
+
+    .react-js-cron-minutes > span {
+      padding-left: ${theme.gridUnit}px;
+    }
+
+    div:has(.react-js-cron-hours) {
+      width: 100%;
+    }
+
+    :not(div:has(.react-js-cron-hours)) {
+      display: flex;
+      flex-wrap: nowrap;
+    }
+
+    .react-js-cron-select.ant-select {
+      width: 100%;
+      .ant-select-selector {
+        flex-wrap: nowrap;
+      }
+    }
+
+    .react-js-cron-field {
+      width: 100%;
+      margin-bottom: 0px;
+      > span {
+        margin-left: 0px;
+      }
+    }
+
+    .react-js-cron-custom-select .ant-select-selection-placeholder {
+      flex: auto;
+      border-radius: ${theme.gridUnit}px;
+    }
+
+    .react-js-cron-custom-select .ant-select-selection-overflow-item {
+      align-self: center;
+    }
+
+    .react-js-cron-select > div:first-of-type,
+    .react-js-cron-custom-select {
+      border-radius: ${theme.gridUnit}px;
+    }
+  `}
 `;
```

Comment:
Need to look at the styling here again, it is mysteriously rendering with all labels on separate lines from the inputs, without any changes I've made since the last correction